src/ecalc_cli/infrastructure/file_resource_service.py

```diff
@@ -0,0 +1,43 @@
+from pathlib import Path
+from typing import Callable, Dict
+
+from libecalc.common.errors.exceptions import EcalcError, HeaderNotFound
+from libecalc.common.logger import logger
+from libecalc.infrastructure.file_io import read_facility_resource, read_timeseries_resource
+from libecalc.presentation.yaml.resource import Resource
+from libecalc.presentation.yaml.resource_service import ResourceService
+from libecalc.presentation.yaml.yaml_entities import MemoryResource
+from libecalc.presentation.yaml.yaml_models.yaml_model import YamlValidator
+
+
+class FileResourceService(ResourceService):
```

Comment:
What if we were to provide a Resource that is not loaded into memory - but that reads from the file on the fly, how would we name the service responsible for that?

Reply:
StreamResourceService :eyes: 

Not sure about the naming. `MemoryResource` was something I decided to use after I created this one. Maybe `MemoryResourceService` works better. Not sure what to call the different implementations. Might be easier to name them when we have more.